src/contracts/signing-key-events-cache/signing-key-events-cache.service.ts

```diff
@@ -70,12 +76,18 @@ export class SigningKeyEventsCacheService {
       this.getCachedEvents(),
     ]);
 
+    // TODO: do we need to check cache validity ?
+
     const firstNotCachedBlock = initialCache.headers.endBlock + 1;
     const toBlock = latestBlock - SIGNING_KEYS_EVENTS_CACHE_LAG_BLOCKS;
 
     const totalEventsCount = initialCache.data.length;
     let newEventsCount = 0;
 
+    const stakingModulesContracts =
+      await this.repositoryService.getCachedStakingModulesContracts();
+    const stakingModulesAddresses = Object.keys(stakingModulesContracts);
+
```

Comment:
Should we replace this call with `getStakingModules`? As I see it, this function does the same job

Reply:
And should we re-check the list of modules in each `getUpdatedSigningKeyEvents` call?

Reply:
I don't think so, as we check it in handleNewBlock and use getUpdatedSigningKeyEvents in the same iteration to search for duplicates.